### PR TITLE
refactor(rpc): Mechanical/Manual RPC backports 

### PR DESCRIPTION
**Pull request by Munkybooty**

### CODE REVIEW

**Comment by Munkybooty** [comment]
I need to clean up [27a055d](https://github.com/dashpay/dash/pull/4734/commits/27a055d710c54970a60cab773d50634fbade42ee), the commit message should say why, but It does need to be fixed, as per Pasta's request i will batch the changes for easier review

**Comment by Munkybooty** [comment]
I honestly am unsure of what all is needed to fix the breakages
@UdjinM6 any thoughts?

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
Well.. some parts are missing and some changes are excessive :) You also need to revert few things and adjust dash-specific rpcs to make them compatible with `Check()`. Pls see https://github.com/UdjinM6/dash/commits/pr4734. (these changes are huge so pretty sure I messed up somewhere too, pls double check/test)

**Comment by Munkybooty** [comment]
So I implemented the changes and feature_llmq_rotations has just decided to die

**Comment by PastaPastaPasta** [comment]
Well, it seems like you messed up the `quorum` rpc command. See 
```
2022-04-16T20:50:22.877000Z TestFramework (ERROR): JSONRPC error
Traceback (most recent call last):
  File "/builds/dashpay/dash/build-ci/dashcore-linux32/test/functional/test_framework/test_framework.py", line 234, in main
    self.run_test()
  File "/builds/dashpay/dash/build-ci/dashcore-linux32/test/functional/feature_llmq_rotation.py", line 67, in run_test
    (quorum_info_0_0, quorum_info_0_1) = self.mine_cycle_quorum(llmq_type_name=llmq_type_name, llmq_type=llmq_type)
  File "/builds/dashpay/dash/build-ci/dashcore-linux32/test/functional/test_framework/test_framework.py", line 1456, in mine_cycle_quorum
    quorum_rotation_info = self.nodes[0].quorum("rotationinfo", best_block_hash, 0, False)
  File "/builds/dashpay/dash/build-ci/dashcore-linux32/test/functional/test_framework/coverage.py", line 47, in __call__
    return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
  File "/builds/dashpay/dash/build-ci/dashcore-linux32/test/functional/test_framework/authproxy.py", line 144, in __call__
    raise JSONRPCException(response['error'], status)
test_framework.authproxy.JSONRPCException: quorum "command"
Set of commands for quorums/LLMQs.
To get help on individual commands, use "help quorum command".

Available commands:
  list              - List of on-chain quorums
  info              - Return information about a quorum
  dkgsimerror       - Simulates DKG errors and malicious behavior
  dkgstatus         - Return the status of the current DKG process
  memberof          - Checks which quorums the given masternode is a member of
  sign              - Threshold-sign a message
  verify            - Test if a quorum signature is valid for a request id and a message hash
  hasrecsig         - Test if a valid recovered signature is present
  getrecsig         - Get a recovered signature
  isconflicting     - Test if a conflict exists
  selectquorum      - Return the quorum that would/should sign a request
  getdata           - Request quorum data from other masternodes in the quorum
  rotationinfo      - Request quorum rotation information

Arguments:
1. command    (string, required) The command to execute
 (-1)
2022-04-16T20:50:22.929000Z TestFramework (INFO): Stopping nodes
2022-04-16T20:50:23.596000Z TestFramework (WARNING): Not cleaning up dir /builds/dashpay/dash/build-ci/dashcore-linux32/testdatadirs/test_runner_âˆ‹_ðŸƒ_20220416_214406/feature_llmq_rotation_131
2022-04-16T20:50:23.596000Z TestFramework (ERROR): Test failed. Test logging available at /builds/dashpay/dash/build-ci/dashcore-linux32/testdatadirs/test_runner_âˆ‹_ðŸƒ_20220416_214406/feature_llmq_rotation_131/test_framework.log
```

**Comment by Munkybooty** [comment]
I wanna say I fixed it but now `p2p_instantsend.py` is broken, it's progress though! One thing fixed and another thing broke

**Comment by UdjinM6** [comment]
> I wanna say I fixed it but now `p2p_instantsend.py` is broken, it's progress though! One thing fixed and another thing broke

this should be fixed by #4790, pls rebase

**Comment by UdjinM6** [comment]
pls see ff954e0add

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/rpc/rpcquorums.cpp [hunk 1]
```diff
@@ -727,31 +706,25 @@ static UniValue quorum_rotationinfo(const JSONRPCRequest& request)
...
+        return quorum_memberof(new_request);
+    } else if (command == "quorumsign" || command == "quorumverify" || command == "quorumhasrecsig" || command == "quorumgetrecsig" || command == "quorumisconflicting") {
+        return quorum_sigs_cmd(new_request);
+    } else if (command == "quorumselectquorum") {
+        return quorum_selectquorum(new_request);
+    } else if (command == "quorumdkgsimerror") {
+        return quorum_dkgsimerror(new_request);
+    } else if (command == "quorumgetdata") {
+        return quorum_getdata(new_request);
     } else if (command == "rotationinfo") {
```
Comment (UdjinM6) [inline src/rpc/rpcquorums.cpp:728]:
```suggestion
    } else if (command == "quorumrotationinfo") {
```

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

(sooo many conflicting PRs...)

**Comment by Munkybooty** [comment]
Totally forgot this thing has sat as a draft this whole time

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

time to make everything else need rebase :D

**Comment by Munkybooty** [comment]
Panic Mode

**Pull request merged**